codegenerator/cli/templates/static/codegen/src/eventFetching/FetchState.res

```diff
@@ -1028,156 +751,82 @@ let pruneDynamicContractAddressesFromFirstChangeEvent = (
     )
-  {
-    ...self,
-    pendingDynamicContracts,
-    baseRegister,
-  }
+
+  fetchState->updateInternal(~pendingDynamicContracts, ~registers)
 }
 
 /**
 * Returns a boolean indicating whether the fetch state is actively indexing
 * used for comparing event queues in the chain manager
 */
-let isActivelyIndexing = ({baseRegister} as fetchState: t, ~endBlock) => {
-  // nesting to limit additional unnecessary computation
-  switch (baseRegister.registerType, endBlock) {
-  | (RootRegister, Some(endBlock)) =>
-    let isPastEndblock = getLatestFullyFetchedBlock(fetchState).blockNumber >= endBlock
-    if isPastEndblock {
-      baseRegister->registerQueueSize > 0
-    } else {
-      true
-    }
-  | _ => true
-  }
-}
-
-let getNumContracts = (self: t) => {
-  let rec loop = (register: register, ~accum=0) => {
-    let accum = accum + register.contractAddressMapping->ContractAddressingMap.addressCount
-    switch register.registerType {
-    | RootRegister => accum
-    | DynamicContractRegister({nextRegister}) => nextRegister->loop(~accum)
+let isActivelyIndexing = (
+  {mostBehindRegister, pendingDynamicContracts} as fetchState: t,
+  ~endBlock,
+) => {
+  if pendingDynamicContracts->Utils.Array.isEmpty {
+    switch endBlock {
+    | Some(endBlock) =>
+      let isPastEndblock = mostBehindRegister.latestFetchedBlock.blockNumber >= endBlock
+      if isPastEndblock {
+        fetchState->queueSize > 0
+      } else {
+        true
+      }
+    | None => true
     }
+  } else {
+    true
   }
-  loop(self.baseRegister)
 }
 
-/**
-Helper functions for debugging and printing
-*/
-module DebugHelpers = {
-  let registerToString = register =>
-    switch register {
-    | RootRegister => "root"
-    | DynamicContractRegister({id: {blockNumber, logIndex}}) =>
-      `DC-${blockNumber->Int.toString}-${logIndex->Int.toString}`
-    }
-
-  let rec getQueueSizesInternal = (register: register, ~accum) => {
-    let next = (register.registerType->registerToString, register.fetchedEventQueue->Array.length)
-    let accum = list{next, ...accum}
-    switch register.registerType {
-    | RootRegister => accum
-    | DynamicContractRegister({nextRegister}) => nextRegister->getQueueSizesInternal(~accum)
-    }
-  }
-
-  let getQueueSizes = (self: t) =>
-    self.baseRegister->getQueueSizesInternal(~accum=list{})->List.toArray->Js.Dict.fromArray
-
-  let rec numberRegistered = (~accum=0, self: register) => {
-    let accum = accum + 1
-    switch self.registerType {
-    | RootRegister => accum
-    | DynamicContractRegister({nextRegister}) => nextRegister->numberRegistered(~accum)
-    }
-  }
-
-  let rec getRegisterAddressMaps = (self: register, ~accum=[]) => {
-    accum->Js.Array2.push(self.contractAddressMapping.nameByAddress)->ignore
-    switch self.registerType {
-    | RootRegister => accum
-    | DynamicContractRegister({nextRegister}) => nextRegister->getRegisterAddressMaps(~accum)
-    }
+// FIXME: Should include pending contracts?
```

Comment:
Yes it definitely should!

Reply:
I didn't fix it since I planned to change the way we split partitions